pallets/creator-staking/src/types/mod.rs

```diff
@@ -0,0 +1,202 @@
+// In structs which are bound in size, `MaxEncodedLen` can just be derived but that's not the case
+// for standard `vec`. To fix this 100% correctly, we'd need to do one of the following:
+//
+// - Use `BoundedVec` instead of `Vec` and do storage migration
+// - Introduce a new type `S: Get<u32>` into the aforementioned structs and use it to inject max
+//   allowed size, thus allowing us to correctly calculate max encoded len
+//
+// The issue with first approach is that it requires storage migration which we want to avoid
+// unless it's really necessary. The issue with second approach is that it makes code much more
+// difficult to work with since all of it will be ridden with injections of the `S` type.
+const MAX_ASSUMED_VEC_LEN: u32 = 10;
+
+// TODO: there are few more options to chose from:
+//     - UnregistrationSource: This emphasizes where the unregistration is coming from.
+//     - UnregisteringParty: This focuses on who is performing the unregistration.
+//     - UnregistrationActor: This could signify the entity taking action in the unregistration process.
+//     - UnregistrationInitiator: This implies the entity that started or initiated the unregistration.
+/// This enum is used to determine who is calling the `unregister_creator` function.
+pub(super) enum UnregistrationAuthority<AccountId> {
+    Creator(AccountId),
+    Root,
+}
+
+/// Creator State descriptor
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub(super) enum CreatorState {
+    /// Creator is registered and active.
+    Registered,
+    /// Creator has been unregistered and is inactive.
+    /// Claim for past eras and unbonding is still possible but no additional staking can be done.
+    Unregistered(EraIndex),
+}
+
+#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct CreatorInfo<AccountId> {
+    /// Space owner account
+    pub(super) stakeholder: AccountId,
+    /// Current Creator State
+    pub(super) state: CreatorState,
+}
+
+/// Used to split total EraPayout among creators.
+/// Each tuple (creator, era) has this structure.
+/// This will be used to reward creators developer and his stakers.
+#[derive(Clone, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct CreatorStakeInfo<Balance: HasCompact + MaxEncodedLen> {
+    /// Total staked amount.
+    #[codec(compact)]
+    pub total: Balance,
+    /// Total number of active stakers
+    #[codec(compact)]
+    pub(super) number_of_stakers: u32,
+    /// Indicates whether rewards were claimed for this era or not
+    pub(super) creator_reward_claimed: bool,
+}
+
+/// Contains information about account's locked & unbonding balances.
+#[derive(Clone, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct AccountLedger<Balance: AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen> {
```

Comment:
Can we call it `StakerLedger` to make it more specific? Do we use it with creator in any way? 
```suggestion
pub struct StakerLedger<Balance: AtLeast32BitUnsigned + Default + Copy + MaxEncodedLen> {
```